platform-tests/src/test/java/org/junit/platform/suite/commons/SuiteLauncherDiscoveryRequestBuilderTests.java

```diff
@@ -214,6 +217,88 @@ class Suite {
 		assertEquals(TestCase.class, exactlyOne(selectors).getJavaClass());
 	}
 
+	@Test
+	void selectMethodsOnlyByName() {
+		class TestClass {
+			public void testMethod() {
+			}
+
+			public void anotherTestMethod() {
+			}
+		}
+		@SelectMethods({ @SelectMethod(name = "testMethod"), @SelectMethod(name = "anotherTestMethod") })
+		class Suite {
+		}
+		LauncherDiscoveryRequest request = builder.suite(Suite.class).build();
+		List<MethodSelector> selectors = request.getSelectorsByType(MethodSelector.class);
+		assertEquals(2, selectors.size());
+		assertTrue(selectors.stream().anyMatch(selector -> selector.getMethodName().equals("testMethod")));
+		assertTrue(selectors.stream().anyMatch(selector -> selector.getMethodName().equals("anotherTestMethod")));
+	}
```

Comment:
You've got two problems going on here.

1. Your assertions are quite complex. They're quite complex because your test case is quite complex.
  
   Start with the simplest case of having a single `@SelectMethod` annotation for a single class with a single zero-arg method. Then from that expand in different directions of complexity. Multiple methods, single method with one argument, two arguments. 

   To make this easier you can use the `exactlyOne` to unpack the lists into a single object.

2. Your assertions do not verify that all 3 attributes were copied over into the `MethodSelector`. You should verify the class name, method name, and parameter types.